### PR TITLE
Past events feed

### LINKED ISSUE #1188: Add past events RSS feed support
### Is your enhancement related to a problem? Please describe.

Currently, the GatherPress events RSS feed only shows upcoming events. Users cannot easily subscribe to past events via RSS, which limits their ability to stay informed about completed events or review event history through their RSS reader.

We are extending the changes already made in #901 to fix this issue.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jmarx**
### Description of the Change

Closes #1188

### How to test the Change
**Test the past events feed:**
Visit http://yoursite.com/event/feed/?type=past - should show past events
Visit http://yoursite.com/event/feed/ - should show upcoming events
**Test the automatic feed link on past events page:**
Configure a past events archive page in GatherPress Settings
Visit that page and check the RSS feed link in page source
Verify the link includes ?type=past parameter
Test edge cases:
?type=invalid should default to upcoming events
?type= should default to upcoming events

### Changelog Entry
Adds support for past events RSS feeds via ?type=past parameter. When viewing the past events archive page, the RSS feed link automatically includes this parameter.
Added - Past events RSS feed support with ?type=past parameter

### Credits

Props @jmarx 

### Checklist:

- [X] I agree to follow this project's [**Code of Conduct**](https://github.com/GatherPress/gatherpress/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my change.
- [X] All new and existing tests pass.

### CODE REVIEW

File: includes/core/classes/class-feed.php [hunk 1]
```diff
@@ -63,13 +63,17 @@ protected function setup_hooks(): void {
 
 		// Hook into the main query to handle events feeds.
 		add_action( 'pre_get_posts', array( $this, 'handle_events_feed_query' ) );
+
+		// Modify feed link for past events page.
+		add_filter( 'post_type_archive_feed_link', array( $this, 'modify_feed_link_for_past_events' ), 10, 1 );
```
Comment (mauteri) [inline includes/core/classes/class-feed.php]: Don't need `10, 1` as those are defaults.

File: includes/core/classes/class-event-query.php [hunk 2]
```diff
@@ -189,8 +189,9 @@ public function prepare_event_query_before_execution( WP_Query $query ): void {
 					if ( $page->id === $query->queried_object_id ) {
 						$query->set( 'post_type', 'gatherpress_event' );
 
-						$page_id                     = $query->queried_object_id;
-						$events_query                = $key;
+						$page_id      = $query->queried_object_id;
+						$events_query = $key;
```
Comment (mauteri) [inline includes/core/classes/class-event-query.php:199]: Add a line break after setting variables

File: includes/core/classes/class-event-query.php [hunk 3]
```diff
@@ -189,8 +189,9 @@ public function prepare_event_query_before_execution( WP_Query $query ): void {
 					if ( $page->id === $query->queried_object_id ) {
 						$query->set( 'post_type', 'gatherpress_event' );
```
Comment (mauteri) [inline includes/core/classes/class-event-query.php]: Maybe bring this down with the other `->set` on line 194 to keep like things together.

File: includes/core/classes/class-feed.php [hunk 4]
```diff
@@ -94,7 +98,15 @@ public function handle_events_feed_query( WP_Query $query ): void {
 			if ( str_contains( $request_uri, '/' . $rewrite_slug . '/' . $GLOBALS['wp_rewrite']->feed_base ) ) {
 				// Set the post type and let Event_Query handle the rest.
 				$query->set( 'post_type', Event::POST_TYPE );
-				$query->set( 'gatherpress_events_query', 'upcoming' );
+
+				// Check for type parameter to determine if we want past or upcoming events.
+				$event_type = 'upcoming'; // Default to upcoming events.
```
Comment (mauteri) [inline includes/core/classes/class-feed.php]: Add a line break here for readability.

File: includes/core/classes/class-feed.php [hunk 5]
```diff
@@ -94,7 +98,15 @@ public function handle_events_feed_query( WP_Query $query ): void {
...
+				if ( isset( $_GET['type'] ) && 'past' === sanitize_text_field( wp_unslash( $_GET['type'] ) ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Public feed URL parameter, nonce not required.
```
Comment (mauteri) [inline includes/core/classes/class-feed.php]:
```
if ( 
    isset( $_GET['type'] ) && 
    'past' === sanitize_text_field( wp_unslash( $_GET['type'] ) ) 
) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Public feed URL parameter, nonce not required.
```
multiline for readability.

File: includes/core/classes/class-feed.php [hunk 6]
```diff
@@ -302,4 +314,22 @@ public function apply_event_content( string $content ): string {
 		 */
 		return apply_filters( 'gatherpress_event_feed_content', $content );
 	}
+
+	/**
+	 * Modify feed link for past events page.
+	 *
```
Comment (mauteri) [inline includes/core/classes/class-feed.php:324]: Add `@since 1.0.0`

File: includes/core/classes/class-feed.php [hunk 7]
```diff
@@ -302,4 +314,22 @@ public function apply_event_content( string $content ): string {
...
+	 * @param string $feed_link The feed link URL.
+	 * @return string The modified feed link URL.
+	 */
+	public function modify_feed_link_for_past_events( $feed_link ): string {
+		global $wp_query;
+
+		// Check if we're on the past events page by looking for the gatherpress_events_query var.
+		if ( isset( $wp_query->query_vars['gatherpress_events_query'] ) && 'past' === $wp_query->query_vars['gatherpress_events_query'] ) {
```
Comment (mauteri) [inline includes/core/classes/class-feed.php]: Same here, multiline for readability.

**Review by mauteri — CHANGES_REQUESTED** [review]

File: includes/core/classes/class-event-query.php [hunk 8]
```diff
@@ -187,10 +187,11 @@ public function prepare_event_query_before_execution( WP_Query $query ): void {
...
 
 					if ( $page->id === $query->queried_object_id ) {
-						$query->set( 'post_type', 'gatherpress_event' );
+						$page_id      = $query->queried_object_id;
+						$events_query = $key;
 
-						$page_id                     = $query->queried_object_id;
-						$events_query                = $key;
+						$query->set( 'post_type', 'gatherpress_event' );
+						$query->set( 'gatherpress_events_query', $key );
```
Comment (mauteri) [inline includes/core/classes/class-event-query.php]: Let's change to `gatherpress_event_query`

File: includes/core/classes/class-feed.php (hunk 6 above)
Comment (jmarx) [reply includes/core/classes/class-feed.php:324]: @mauteri It's already there. On line 325

File: includes/core/classes/class-event-query.php (hunk 8 above)
Comment (jmarx) [reply includes/core/classes/class-event-query.php]: @mauteri I just realized we can't make this change without changing a bunch of things. I feel like we should keep it plural for now. Or I will get to that change later in the week where I can test the query loop block which uses this. Up to you.

Comment (jmarx) [reply includes/core/classes/class-event-query.php]: @mauteri Just to follow up on this. I reverted everything back to plural for now because it was indeed messing up the `Event Query` block. I am going to troubleshoot and try to fix that(simple search and replace/`npm run build` did not seem to do the trick).  But for now, I am going to revert this change so the pr is in a working state. But I'll keep working on it.

Comment (jmarx) [reply includes/core/classes/class-event-query.php]: @mauteri OK, it should be all set now. Please review again.

File: includes/core/classes/class-event-query.php (hunk 2 above)
Comment (mauteri) [reply includes/core/classes/class-event-query.php:199]: @jmarx just add this line break for clear separation and readability.

Comment (jmarx) [reply includes/core/classes/class-event-query.php:199]: @mauteri that line break is already there?
Screenshot 2025-09-22 at 8 25 21 PM
Screenshot 2025-09-22 at 8 24 47 PM

File: includes/core/classes/class-event-query.php [hunk 9]
```diff
@@ -187,10 +187,11 @@ public function prepare_event_query_before_execution( WP_Query $query ): void {
 					$page = $value[0];
 
 					if ( $page->id === $query->queried_object_id ) {
-						$query->set( 'post_type', 'gatherpress_event' );
+						$page_id      = $query->queried_object_id;
+						$events_query = $key;
```
Comment (jmarx) [inline includes/core/classes/class-event-query.php:200]: @mauteri this is the line break you want? Am I missing something?

**Review by mauteri — APPROVED** [review]
